src/Lightning.ts

```diff
@@ -263,12 +263,21 @@ export const LightningMixin = (superclass) => class extends superclass {
   async pay(params: IPaymentRequest): Promise<payInvoiceResult | Error> {
     let lightningLogger = this.logger.child({ topic: "payment", protocol: "lightning", transactionType: "payment" })
 
+
     const { tokens, mtokens, username: input_username, destination, pushPayment, id, routeHint, messages, memoInvoice, payment, cltv_delta, features, max_fee } = await this.validate(params, lightningLogger)
-    const { memo: memoPayer } = params
+    const { memo: memoPayer, twoFactorToken: token } = params
 
     // not including message because it contains the preimage and we don't want to log this
     lightningLogger = lightningLogger.child({ decoded: { tokens, destination, pushPayment, id, routeHint, memoInvoice, memoPayer, payment, cltv_delta, features }, params })
 
+    if(yamlConfig.twoFactor?.enabled && this.user.twoFactor.secret && tokens > this.user.twoFactor.threshold) {
```

Comment:
maybe we should have a default in `schema.ts` for twoFactor instead of having to use a `?` in `yamlConfig.twoFactor?.enabled`

Reply:
We can have a default in the dbMetadata schema but I don't think it's the right place for this to be. I was thinking about the `twoFactor.enabled` in a similar way to how we are doing `proxyChecking.enabled` and thus yamlConfig seemed the right place for it. We can of course remove this check altogether in the future and just rely on whether the user's object has a secret or not (in which case there won't be a system-wide twofactor enable/disable flag)